AU/Public/Test-Package.ps1

```diff
@@ -78,9 +95,103 @@ function Test-Package {
     Write-Host "  Name:".PadRight(15)      $package_name
     Write-Host "  Version:".PadRight(15)   $package_version
     if ($Parameters) { Write-Host "  Parameters:".PadRight(15) $Parameters }
-    if ($Vagrant)    { Write-Host "  Vagrant: ".PadRight(15) $Vagrant }
+
+    if ($Sandbox) {
+        if (-not (Get-Command WindowsSandbox -ErrorAction SilentlyContinue)) {
+            Write-Error "Windows Sandbox is not available. Please try Local or Vagrant methods." -ErrorAction Stop
+        }
+        Write-Host "`nTesting package using Windows Sandbox"
+
+        $SandboxTempFolder = Join-Path $env:TEMP "$(New-Guid)"
+        $null = New-Item -Path $SandboxTempFolder\ChocoTestingSandbox.wsb -Force -Value @"
+            <Configuration>
+                <Networking>Enable</Networking>
+                <MappedFolders>
+                    <MappedFolder>
+                        <HostFolder>$($SandboxTempFolder)</HostFolder>
+                        <SandboxFolder>C:\packages\</SandboxFolder>
+                        <ReadOnly>false</ReadOnly>
+                    </MappedFolder>
+                </MappedFolders>
+                <LogonCommand>
+                    <Command>C:\packages\Test-Package.cmd</Command>
+                </LogonCommand>
+            </Configuration>
+"@
+
+        $TestCommand = @(
+            'Start-Transcript -Path c:\packages\log.txt'
+            '$Results = @{}'
+            if ($Install) {
+                "choco install -y $package_name --version $package_version --no-progress --source `"'C:\packages\;https://community.chocolatey.org/api/v2/'`"$(if ($Parameters) {[string]::format(' --packageParameters="''{0}''"', $Parameters)}) | Write-Host"
+                '$Results.InstallExitCode = $LASTEXITCODE'
+            }
+            if ($Uninstall) {
+                "choco uninstall -y $package_name | Write-Host"
+                '$Results.UninstallExitCode = $LASTEXITCODE'
+            }
+            'Stop-Transcript'
+            'Copy-Item C:\ProgramData\chocolatey\logs\* -Destination C:\packages\'
+            '$Results | Export-Clixml -Path C:\packages\results.clixml'
+        ) -join '; '
+        $null = New-Item -Path $SandboxTempFolder\Test-Package.cmd -Value @"
+            @"%SystemRoot%\System32\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -InputFormat None -ExecutionPolicy Bypass -Command "iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))" && SET "PATH=%PATH%;%ALLUSERSPROFILE%\chocolatey\bin"
+            @"%SystemRoot%\System32\WindowsPowerShell\v1.0\powershell.exe" -NoProfile -InputFormat None -ExecutionPolicy Bypass -Command "$($TestCommand)"
+"@
+        Copy-Item -Path $Nu -Destination $SandboxTempFolder
+
+        try {
+            & WindowsSandbox "$SandboxTempFolder\ChocoTestingSandbox.wsb"
```

Comment:
We should check for an existing windowssandbox process. If there is already one, then this will fail and we'll be waiting for the timeout to hit needlessly.

![image](https://github.com/chocolatey-community/chocolatey-au/assets/30301021/a2c7e554-3289-4bca-b7a1-8e7a5de92b08)

I would recommend the check happen early before we've done a bunch of setup.